lib/matplotlib/text.py

```diff
@@ -455,6 +457,8 @@ def _get_layout(self, renderer):
                 offsety = ymax1
             elif valign == 'baseline':
                 offsety = ymax1 - baseline
+            elif valign == 'center_baseline':
+                offsety = (ymin1 + ymax1) / 2.0
```

Comment:
This was wrong, actually (and not addressed by existing tests, since it's a new vertical alignement type).  Fixed now.